pandas/io/json/_normalize.py

```diff
@@ -261,9 +261,12 @@ def _pull_records(js: Dict[str, Any], spec: Union[List, str]) -> List:
 
     if isinstance(data, list) and not data:
         return DataFrame()
-
-    # A bit of a hackjob
-    if isinstance(data, dict):
+    elif isinstance(data, abc.Iterator):
+        # GH35923 Fix pd.json_normalize to not skip the first element of a
+        # generator input
+        data = list(data)
+    elif isinstance(data, dict):
+        # A bit of a hackjob
```

Comment:
does *anything* hit an else here? (e.g. put one and put an exception NotImplementedError) and see

Reply:
sorry you need an `elif abc.Iterable` i think maybe

Reply:
The `Iterator` should only capture the generator input
```python
In [1]: from collections.abc import Iterator

In [2]: isinstance([1,2,3], Iterator)
Out[2]: False

In [3]: isinstance({"a":1}, Iterator)
Out[3]: False

In [4]: def gen():
   ...:     yield from [1,2,3]
   ...:

In [5]: isinstance(gen(), Iterator)
Out[5]: True
```

I can also change `Iterator` to `Iterable`, then I do have to move the `dict` check to be before the `Iterable` check. Thats also fine.

Reply:
sure, the goal is to lock down the acceptable types, generally Iterable/Dictlike are ok, then *anything* else should raise (we may or may not have tests for this part). I think its worth fixing here to the extent you are changing anyhow.